### PR TITLE
feat: support multiple idps

### PR DESCRIPTION
**Pull request by astsiapanay**

### CODE REVIEW

File: src/test/java/com/epam/aidial/core/security/AccessTokenValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        when(provider2.extractClaims(any(DecodedJWT.class), eq(true))).thenReturn(Future.succeededFuture(new ExtractedClaims("sub", Collections.emptyList(), "hash")));
+        List<IdentityProvider> providerList = List.of(provider1, provider2);
+        validator.setProviders(providerList);
+        KeyPair keyPair = generateRsa256Pair();
+        Algorithm algorithm = Algorithm.RSA256((RSAPublicKey) keyPair.getPublic(), (RSAPrivateKey) keyPair.getPrivate());
+        String token = JWT.create().withClaim("iss", "issuer2").sign(algorithm);
+        Future<ExtractedClaims> future = validator.extractClaims(getBearerHeaderValue(token), true);
+        assertNotNull(future);
+        future.onComplete(res -> {
+            assertTrue(res.succeeded());
```
Comment (Maxim-Gadalov) [inline src/test/java/com/epam/aidial/core/security/AccessTokenValidatorTest.java:130]: can you assert that result `ExtractedClaims` object match expected one

Comment (astsiapanay) [reply src/test/java/com/epam/aidial/core/security/AccessTokenValidatorTest.java:130]: fixed

**Review by Maxim-Gadalov — DISMISSED** [review]

File: src/main/java/com/epam/aidial/core/security/IdentityProvider.java [hunk 2]
```diff
@@ -49,6 +50,8 @@ public class IdentityProvider {
 
     private final long negativeCacheExpirationMs;
 
+    private Pattern issuerPattern;
```
Comment (artsiomkorzun) [inline src/main/java/com/epam/aidial/core/security/IdentityProvider.java]: final?

Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/security/IdentityProvider.java]: fix

**Review by artsiomkorzun — DISMISSED** [review]

**Comment by astsiapanay** [comment]
/deploy-review
>Deployment status: success

File: README.md [hunk 3]
```diff
@@ -27,26 +27,28 @@ Static settings are used on startup and cannot be changed while application is r
...
-| storage.identity                     | -                  |Blob storage access key
-| storage.credential                   | -                  |Blob storage secret key
-| storage.bucket                       | -                  |Blob storage bucket
-| storage.createBucket                 | false              |Indicates whether bucket should be created on start-up
+| Setting                                  | Default            |Description
+|------------------------------------------|--------------------|-
+| config.files                             | aidial.config.json |Config files with parts of the whole config.
+| config.reload                            | 60000              |Config reload interval in milliseconds.
+| identityProviders                        | -                  |List of identity providers
+| identityProviders[*].jwksUrl             | -                  |Url to jwks provider.
```
Comment (Maxim-Gadalov) [inline README.md]: fix settings pattern since it's not an array any more

File: src/main/java/com/epam/aidial/core/security/IdentityProvider.java [hunk 4]
```diff
@@ -94,7 +105,7 @@ private List<String> extractUserRoles(DecodedJWT token) {
             return token.getClaim(rolePath[0]).asList(String.class);
         }
         Map<String, Object> claim = token.getClaim(rolePath[0]).asMap();
-        for (int i = 1; i < rolePath.length; i++) {
+        for (int i = 1; claim != null && i < rolePath.length; i++) {
```
Comment (Maxim-Gadalov) [inline src/main/java/com/epam/aidial/core/security/IdentityProvider.java]: `claim != null` - this check makes sense only for line `Map<String, Object> claim = token.getClaim(rolePath[0]).asMap();`
we can check it before loop and return result immediately, but this is a personal preference

Comment (astsiapanay) [reply src/main/java/com/epam/aidial/core/security/IdentityProvider.java]: fixed

File: README.md (hunk 3 above)
Comment (astsiapanay) [reply README.md]: fixed

**Review by Maxim-Gadalov — APPROVED** [review]

**Pull request merged**